roiextractors/extraction_tools.py

```diff
@@ -39,3 +43,92 @@ def get_video_shape(video):
         num_channels, num_frames, size_x, size_y = video.shape
 
     return num_channels, num_frames, size_x, size_y
+
+
+def check_get_frames_args(func):
+    @wraps(func)
+    def corrected_args(*args, **kwargs):
+        # parse args and kwargs
+        assert len(args) >= 2, "'get_frames' requires 'frame_idxs' as first argument"
+        if len(args) == 2:
+            imaging = args[0]
+            frame_idxs = args[1]
+            channel = kwargs.get('channel', 0)
+        elif len(args) == 3:
+            imaging = args[0]
+            frame_idxs = args[1]
+            channel = args[2]
+        else:
+            raise Exception("Too many arguments!")
+
+        channel = int(channel)
+        if isinstance(frame_idxs, (int, np.integer)):
+            frame_idxs = [frame_idxs]
+        frame_idxs = np.array(frame_idxs)
+        assert np.all(frame_idxs < imaging.get_num_frames()), "'frame_idxs' exceed number of frames"
+        kwargs['channel'] = channel
+        get_frames_correct_arg = func(args[0], frame_idxs, channel)
+
+        if len(frame_idxs) == 1:
+            return get_frames_correct_arg[0]
+        else:
+            return get_frames_correct_arg
+    return corrected_args
+
+
+def check_get_videos_args(func):
+    @wraps(func)
```

Comment:
Same here. I do not understand why this is necessary. You could just have

```python
def get_video(self, start_frame=None, end_frame=None, channel=0):
    return self._video[channel, start_frame: end_frame]
```
Those `start_frame` and `end_frame` default will be parsed correctly and have the same effect as the current code. The error messages you'd get would be pretty interpretable too.

Reply:
Why even have the `get_video` method? Wouldn't `get_frames(range(start_frame, end_frame))` have the same effect?

Reply:
I think the `get_video` is more user friendly than the `get_frames(range(start_frame, end_frame))`. They are somewhat overlapping, but the use case is different:

- `get_frames` --> used to grab specific frames 
(e.g. random frames to get a mean image)
- `get_video` --> retrieve continuous snippets of video

Regarding the decorator, again, not necessary but useful! We can check that end_frame is not less than start_frame, that the number of frames is not too large, and so on.